### PR TITLE
Non-replicated actors don't need roles adjusting

### PR DESCRIPTION
**Pull request by improbable-danny**
#### Description
Actors which default to having no role currently are considered for replication despite being marked as non-replicated. This is because Unreal uses the RemoteRole to unconsidered network actors (here https://github.com/improbableio/UnrealEngine/blob/4.25-SpatialOSUnrealGDK-release/Engine/Source/Runtime/Engine/Private/NetworkDriver.cpp#L3847). This change also stops these actors having entities and actor channels. 

#### Release note
REQUIRED: Add a release note to the `##Unreleased` section of CHANGELOG.md. You can find guidance for writing useful release notes [here](../SpatialGDK/Extras/internal-documentation/how-to-write-good-release-notes.md). Documentation changes are exempt from this requirement.

#### Tests
NFRs
Unit tests

#### Primary reviewers
@m-samiec @alastairdglennie

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -61,6 +61,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Fixed a crash when spawn requests are forwarded but the `APlayerStart` actor is not resolvable on the target worker.
 - By default, only an Actor's replicated owner hierarchy will be used when determining which worker should have authority over an actor. Non-replicated Actors are now ignored.
 - Fixed a crash that would sometimes occur when connection to SpatialOS fails.
+- Non-replicated actors are no longer considered for replication or migration.
```
Comment (alastairdglennie) [inline CHANGELOG.md:64]:
```suggestion
- Non-replicated Actors are no longer considered for replication or migration. 
```

File: SpatialGDK/Source/SpatialGDK/Private/Interop/GlobalStateManager.cpp [hunk 2]
```diff
@@ -389,14 +389,17 @@ void UGlobalStateManager::HandleActorBasedOnLoadBalancer(AActor* Actor) const
...
+		return;
+	}
+
 	// Replicated level Actors should only be initially authority if:
 	//  - these are workers starting as part of a fresh deployment (tracked by the bCanSpawnWithAuthority bool),
 	//  - these actors are marked as NotPersistent and we're loading from a saved snapshot (which means bCanSpawnWithAuthority is false)
 	//  - the load balancing strategy says this server should be authoritative (as opposed to some other server).
-	bAuthoritative |= (bCanSpawnWithAuthority || Actor->GetClass()->HasAnySpatialClassFlags(SPATIALCLASS_NotPersistent))
-					  && NetDriver->LoadBalanceStrategy->ShouldHaveAuthority(*Actor);
+	bool bAuthoritative = (bCanSpawnWithAuthority || Actor->GetClass()->HasAnySpatialClassFlags(SPATIALCLASS_NotPersistent))
```
Comment (alastairdglennie) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/GlobalStateManager.cpp:401]:
```suggestion
	const bool bAuthoritative = (bCanSpawnWithAuthority || Actor->GetClass()->HasAnySpatialClassFlags(SPATIALCLASS_NotPersistent))
```

**Review by alastairdglennie — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -62,6 +62,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - By default, only an Actor's replicated owner hierarchy will be used when determining which worker should have authority over an actor. Non-replicated Actors are now ignored.
 - Fixed a crash that would sometimes occur when connection to SpatialOS fails.
 - Fixed a crash that occurred when an actor subobject became invalid after applying initial component data.
+- Non-replicated Actors are no longer considered for replication or migration.
```
Comment (improbable-danny) [inline CHANGELOG.md:65]:
```suggestion
- Non-replicated Actors roles are not touched during startup. 
```

Comment (m-samiec) [reply CHANGELOG.md:65]: net roles

**Review by m-samiec — APPROVED** [review]

**Review by mironec — APPROVED** [review]

**Pull request merged**